sema/commons/j2/j2_functions.py

```
@@ -156,6 +157,111 @@
+    # 6. string matching [-]?YYYY for gyear
+    formatted_date = _auto_str_to_formatted_date(str(content), quote)
+    if formatted_date is not None:
+        return formatted_date
+    # 7. int for gyear
+    if isinstance(content, int):
+        return xsd_format_gyear(content, quote)
+    # 8. anything else should raise an error
+    raise ValueError("auto-date format failed to infer date type")
+
+
+def xsd_auto_format_number(content: Any, quote: str, *_: Any) -> str:
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `content`

(ANN401)